### ISSUE TITLE
FTX - KeyError 'symbol'

### ISSUE THREAD

**Issue by Dante2333** [freqtrade/freqtrade#2818]
## Step 1: Have you search for this issue before posting it?

Yes
## Step 2: Describe your environment

  * Operating system: Ubuntu
  * Python Version: 3.7 (`python -V`)
  * CCXT version: latest (`pip freeze | grep ccxt`)
  * Branch:  Develop
  * Last Commit ID: latest
  
## Step 3: Describe the problem:

When running with real funds (not dry run), the bot seems to crash after opening trades.

### Steps to reproduce:

  1. Trade on FTX spot markets and leveraged tokens
  2. Run as usual, wait for buy signal to happen
  3. After buy signal, crashes (it does buy then crashes)
  
### Observed Results:

  * What happened?
The bot bought then crashed
  * What did you expect to happen?
To buy and properly trade.

### Relevant code exceptions or logs:

  ```
2020-01-28 10:59:13,747 - freqtrade.freqtradebot - INFO - Bot heartbeat. PID=22872
2020-01-28 10:59:13,752 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=1, pair=BCHBEAR/USD, amount=0.00569801, open_rate=351.00000000, open_since=2020-01-28 10:59:11)
2020-01-28 10:59:13,960 - freqtrade.commands.trade_commands - INFO - worker found ... calling exit
2020-01-28 10:59:13,960 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': status, 'status': 'process died'}
2020-01-28 10:59:14,349 - freqtrade.freqtradebot - INFO - Cleaning up modules ...
2020-01-28 10:59:14,349 - freqtrade.rpc.rpc_manager - INFO - Cleaning up rpc modules ...
2020-01-28 10:59:15,256 - freqtrade - ERROR - Fatal exception!
Traceback (most recent call last):
  File "/home/dante/freqtrade/freqtrade/main.py", line 36, in main
    return_code = args['func'](args)
  File "/home/dante/freqtrade/freqtrade/commands/trade_commands.py", line 18, in start_trading
    worker.run()
  File "/home/dante/freqtrade/freqtrade/worker.py", line 71, in run
    state = self._worker(old_state=state)
  File "/home/dante/freqtrade/freqtrade/worker.py", line 109, in _worker
    self._throttle(func=self._process, min_secs=throttle_secs)
  File "/home/dante/freqtrade/freqtrade/worker.py", line 122, in _throttle
    result = func(*args, **kwargs)
  File "/home/dante/freqtrade/freqtrade/worker.py", line 132, in _process
    self.freqtrade.process()
  File "/home/dante/freqtrade/freqtrade/freqtradebot.py", line 142, in process
    self.exit_positions(trades)
  File "/home/dante/freqtrade/freqtrade/freqtradebot.py", line 554, in exit_positions
    self.update_trade_state(trade)
  File "/home/dante/freqtrade/freqtrade/freqtradebot.py", line 1032, in update_trade_state
    new_amount = self.get_real_amount(trade, order)
  File "/home/dante/freqtrade/freqtrade/freqtradebot.py", line 1072, in get_real_amount
    trade.open_date)
  File "/home/dante/freqtrade/freqtrade/exchange/common.py", line 113, in wrapper
    return f(*args, **kwargs)
  File "/home/dante/freqtrade/freqtrade/exchange/exchange.py", line 967, in get_trades_for_order
    pair, int((since.replace(tzinfo=timezone.utc).timestamp() - 5) * 1000))
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/ftx.py", line 1070, in fetch_my_trades
    return self.parse_trades(trades, market, since, limit)
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 1591, in parse_trades
    array = [self.extend(self.parse_trade(trade, market), params) for trade in array]
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 1591, in <listcomp>
    array = [self.extend(self.parse_trade(trade, market), params) for trade in array]
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/ftx.py", line 561, in parse_trade
    symbol = market['symbol']
KeyError: 'symbol'
  ```

**Comment by xmatthias** [comment]
This seems to be a issue with CCXT parsing trade data.

Best verify you're using the lastest CCXT version - and try to manually run:

``` python
import ccxt
from datetime import timezone, datetime
ct = ccxt.ftx({<yourkeys>})
since = datetime(2020,1,28, 10, 50,0)
# expect failure in the next line:
trades = ct.fetch_my_trades('<pair>', int((since.replace(tzinfo=timezone.utc).timestamp() - 5) * 1000) )   
print(trades)
```

**Comment by Dante2333** [comment]
I got this error:
```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    trades = ct.fetch_my_trades('<pair>', int((since.replace(tzinfo=timezone.utc).timestamp() - 5) * 1000) )   
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/ftx.py", line 1037, in fetch_my_trades
    market = self.market(symbol)
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 1692, in market
    raise BadSymbol('{} does not have market symbol {}'.format(self.id, symbol))
ccxt.base.errors.BadSymbol: ftx does not have market symbol <pair>
```

**Comment by hroff-1902** [comment]
^^ `<key>` should be replaced with the pair name ("XBT/USD" for example, or "BCHBEAR/USD" which is shown in the logs you posted) existing on the exchange...
as well as `<key>` and `<secrete>` couple of lines above that line...

**Comment by Dante2333** [comment]
Thanks @hroff-1902 
```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    trades = ct.fetch_my_trades('BCHBEAR/USD', int((since.replace(tzinfo=timezone.utc).timestamp() - 5) * 1000) )   
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/ftx.py", line 1070, in fetch_my_trades
    return self.parse_trades(trades, market, since, limit)
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 1591, in parse_trades
    array = [self.extend(self.parse_trade(trade, market), params) for trade in array]
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 1591, in <listcomp>
    array = [self.extend(self.parse_trade(trade, market), params) for trade in array]
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/ftx.py", line 561, in parse_trade
    symbol = market['symbol']
KeyError: 'symbol'
```

**Comment by hroff-1902** [comment]
what is the CCXT version you are using?

**Comment by Dante2333** [comment]
I updated freqtrade through ./setup.sh --update and it seem to be using 1.21.91 but my pip3 shows 1.21.12; I'll re-try with new freqtrade and report back

**Comment by Dante2333** [comment]
After updating to latest freqtrade, still get the error:

```
2020-01-31 03:35:50,341 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=1, pair=BTC/USD, amount=0.00021155, open_rate=9454.00000000, open_since=2020-01-31 03:35:32)
2020-01-31 03:35:50,535 - freqtrade.commands.trade_commands - INFO - worker found ... calling exit
2020-01-31 03:35:50,535 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': status, 'status': 'process died'}
2020-01-31 03:35:50,908 - freqtrade.freqtradebot - INFO - Cleaning up modules ...
2020-01-31 03:35:50,908 - freqtrade.rpc.rpc_manager - INFO - Cleaning up rpc modules ...
2020-01-31 03:36:05,787 - freqtrade - ERROR - Fatal exception!
Traceback (most recent call last):
  File "/home/dante/freqtrade/freqtrade/main.py", line 36, in main
    return_code = args['func'](args)
  File "/home/dante/freqtrade/freqtrade/commands/trade_commands.py", line 20, in start_trading
    worker.run()
  File "/home/dante/freqtrade/freqtrade/worker.py", line 62, in run
    state = self._worker(old_state=state)
  File "/home/dante/freqtrade/freqtrade/worker.py", line 98, in _worker
    self._throttle(func=self._process, min_secs=throttle_secs)
  File "/home/dante/freqtrade/freqtrade/worker.py", line 111, in _throttle
    result = func(*args, **kwargs)
  File "/home/dante/freqtrade/freqtrade/worker.py", line 121, in _process
    self.freqtrade.process()
  File "/home/dante/freqtrade/freqtrade/freqtradebot.py", line 152, in process
    self.exit_positions(trades)
  File "/home/dante/freqtrade/freqtrade/freqtradebot.py", line 571, in exit_positions
    self.update_trade_state(trade)
  File "/home/dante/freqtrade/freqtrade/freqtradebot.py", line 1049, in update_trade_state
    new_amount = self.get_real_amount(trade, order)
  File "/home/dante/freqtrade/freqtrade/freqtradebot.py", line 1089, in get_real_amount
    trade.open_date)
  File "/home/dante/freqtrade/freqtrade/exchange/common.py", line 113, in wrapper
    return f(*args, **kwargs)
  File "/home/dante/freqtrade/freqtrade/exchange/exchange.py", line 967, in get_trades_for_order
    pair, int((since.replace(tzinfo=timezone.utc).timestamp() - 5) * 1000))
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/ftx.py", line 1070, in fetch_my_trades
    return self.parse_trades(trades, market, since, limit)
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 1591, in parse_trades
    array = [self.extend(self.parse_trade(trade, market), params) for trade in array]
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 1591, in <listcomp>
    array = [self.extend(self.parse_trade(trade, market), params) for trade in array]
  File "/home/dante/.local/lib/python3.7/site-packages/ccxt/ftx.py", line 561, in parse_trade
    symbol = market['symbol']
KeyError: 'symbol'
```

**Comment by xmatthias** [comment]
please ensure that the version you're using is the latest one. 

As mentioned above, this is not a freqtrade problem, but most likely one with ccxt - therefore, update ccxt to the latest version (`pip3 install -U ccxt`) - and run the code-snippted i posted above - with your keys - and the pair that has caused the error filled into the code template. (everything in `<>` should be replaced with values matching your environment/trade/keys)


If the error is risen by the code-snippet above - please open an issue about this [in the ccxt repository](https://github.com/ccxt/ccxt/issues/).

**Comment by Dante2333** [comment]
Now it works. I imagine that this update will force freqtrade to use latest ccxt?

**Comment by xmatthias** [comment]
the library is only installed once in your virtual environment.

updating ccxt independently is almost never a problem for freqtrade - so if you're testing new, previously untested exchanges, i'd encourage you to do this more frequently / whenever you encounter a problem where the stack contains `site-packages/ccxt/`.
this will avoid reinvestigating already fixed problems, and will "auto"-fix problems already discovered in ccxt (such as this).

**Comment by Dante2333** [comment]
Thanks, will do, sorry for the inconvenient. I assumed it was already updated on the setupsh file

**Comment by xmatthias** [comment]
in develop, it's updated once per week.

but it requires you to update the bot... that'll not happen automatically.

also - a version newer than 1.21.14 should be available in develop since at least 25 days: #2751
